File: Week 02/id_636/LeetCode_94_636.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        '''
        root: TreeNode
        list[int]
        '''
        # init
        result = []

        def loop(root: TreeNode):
            if not root:
                return

            loop(root.left)
            result.append(root.val)
            loop(root.right)

        loop(root)

        return result

File: Week 02/id_636/LeetCode_94_636.go
package id_636


// Definition for a binary tree node.
type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

func inorderTraversal(root *TreeNode) []int {
	// check root whether root is nil
	if root == nil {
		return nil
	}

	// if root left is nil and right is nil
	// return []
	if root.Left == nil && root.Right == nil {
		return []int{root.Val}
	}

	// result
	result := inorderTraversal(root.Left)
	result = append(result, root.Val)
	result = append(result, inorderTraversal(root.Right)...)

	return result
}
